extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceCallback.kt

```diff
@@ -0,0 +1,78 @@
+package com.mapbox.maps.extension.androidauto
+
+import android.graphics.Rect
+import androidx.car.app.AppManager
+import androidx.car.app.CarContext
+import androidx.car.app.SurfaceCallback
+import androidx.car.app.SurfaceContainer
+import com.mapbox.common.Logger
+import com.mapbox.maps.MapInitOptions
+import com.mapbox.maps.MapboxExperimental
+
+/**
+ * @see MapboxCarMap to create new map experiences.
+ *
+ * This is a Mapbox implementation for the [SurfaceCallback]. It is used to simplify the lower
+ * level calls that manage the map surface. This class handles the surface callbacks and forwards
+ * them to the [CarMapSurfaceOwner] where [MapboxCarMapObserver] instances are notified.
+ */
+@OptIn(MapboxExperimental::class)
+internal class CarMapSurfaceCallback internal constructor(
+  private val carContext: CarContext,
+  private val carMapSurfaceOwner: CarMapSurfaceOwner,
+  private val mapInitOptions: MapInitOptions
+) : SurfaceCallback {
+
+  fun onCreate() {
+    carContext.getCarService(AppManager::class.java).setSurfaceCallback(this)
```

Comment:
how can we guarantee user won't set their own surface callback and overwrite ours? Would be good to mention it somewhere in the documentation.

Reply:
also nit: shall we rename to `onBind()`?

Reply:
> how can we guarantee user won't set their own surface callback and overwrite ours? Would be good to mention it somewhere in the documentation.

Best guarantees I have found, is if we kept a static reference counter to `MapboxCarMap` to ensure there are not multiple instances instantiated. There is not a `getSurfaceCallback`.

In my opinion, error handling here at this time is overkill unless we receive a numerous reports from developers. But I did just add some documentation to the `MapboxCarMap`

```
 * The internals of this class use [AppManager.setSurfaceCallback], which assumes there is a single
 * surface callback. Do not use setSurfaceCallback, and do not create multiple instances of
 * [MapboxCarMap]. 
```